### ISSUE TITLE
File Upload: Add preview/summary for non-text files

### ISSUE THREAD

**Issue by mturley** [patternfly/patternfly-next#2796]
As part of implementing the File Upload component in https://github.com/patternfly/patternfly-react/pull/3865, I was asked how it could be used to upload a non-text file (specifically, a zip file). Since the component includes a textarea for the file contents, the only solution currently is to hide the textarea and provide no preview or summary for an uploaded non-text file, or to have the consumer implement their own custom preview or summary.

Initially I implemented a generic summary for non-text files that could be used by default, but I removed it before merging that PR so we could revisit its design and core implementation first. This is what my initial version looked like:

<img width="624" alt="Screenshot 2020-03-09 14 10 09" src="https://user-images.githubusercontent.com/811963/76244779-1a187d00-6211-11ea-8c94-c987d0f68779.png">

Basically, the idea is to show something useful based on only the data we know about a selected file before reading it: filename, type, size, and last modified date/time.

cc @mcarrano @christiemolloy

**Comment by mcarrano** [comment]
Thanks @mturley . I am going to transfer this to patternfly-design for further consideration.
